src/actionTypes.js

```diff
@@ -222,6 +223,16 @@ type InitialFetchCompleteAction = {|
   type: typeof INITIAL_FETCH_COMPLETE,
 |};
 
+/**
+ * Notify Redux that we've given up on the initial fetch.
+ *
+ * Either because our timeout implementation says we've tried for long
+ * enough, or because the server has responded with a 5xx error.
```

Comment:
> This commit talks about 5xx errors, the previous commit about 4xx errors. Is that correct?

I believe so, although I think I could be clearer. 🙂 Specifically, the role I have in mind for `INITIAL_FETCH_ABORT` is not for cases where breaking from the retry loop is the only rational choice—rather, it's for when we just have a pretty good idea that retrying has stopped being useful, even if that isn't proven. Possibly renaming `INITIAL_FETCH_ABORT` could help here; at least, I should expand the jsdoc.

On a 4xx error, breaking from the retry loop is the only rational choice, and I don't have this action getting dispatched on those. It's a client error; what we've been sending the server isn't any good, and we should bail on this retry loop. We should send a Sentry error report, if the server tells us we've given it garbage; or log out, if the auth is invalid; and so on. I suppose, if we were nimble enough as a client, there might be cases where we have some other reasonable input to try giving the server as a fallback—but I think even in those cases we'd want to throw out the current retry loop and start a new one, with fresh backoff state.

If we've just waited long enough (that we don't think we'll get an answer soon, or that we think the user wants to stop waiting, etc.), that's a suggestion that we should break out of the retry loop, but it's not a proof that waiting longer would be useless. Same with 5xx errors, I think—the server hasn't told us we've done anything wrong, and the issue might clear up in a second or two. But, realistically, what are the chances that it'll be cleared up that soon anyway...might as well give up, even though we might have gotten lucky if we kept trying. 🤷

(I do see that I haven't actually started dispatching `INITIAL_FETCH_ABORT` on 5xx errors in this revision, which is quite old…huh, perhaps I should do that! 🤔)

Reply:
Though, seeing the ratio of the amount I've said just now, above, to what actually made it into the code, I think it could be helpful to have my assumptions examined, possibly on CZO; what do you think? 🙂